src/plugins/data_source_management/public/auth_registry/authentication_methods_registry.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { deepFreeze } from '@osd/std';
+import { EuiSuperSelectOption } from '@elastic/eui';
+
+export interface AuthenticationMethod {
+  name: string;
+  credentialForm: React.JSX.Element;
+  credentialSourceOption: EuiSuperSelectOption<string>;
+}
+
+export type IAuthenticationMethodRegistery = Omit<
+  AuthenticationMethodRegistery,
+  'registerAuthenticationMethod'
+>;
+
+export class AuthenticationMethodRegistery {
+  private readonly authMethods = new Map<string, AuthenticationMethod>();
+  /**
+   * Register a authMethods with function to return credentials inside the registry.
+   * Authentication Method can only be registered once. subsequent calls with the same method name will throw an error.
+   */
+  public registerAuthenticationMethod(method: AuthenticationMethod) {
+    if (this.authMethods.has(method.name)) {
+      throw new Error(`Authentication method '${method.name}' is already registered`);
```

Comment:
just to confirm this will fail the OSD bootstrap right? we should ensure OSD won't start if we see duplicated auth methods

Reply:
Just to confirm here you are not talking about `yarn osd bootstrap`, right? 

`yarn osd bootstrap` is successful. 
`yarn start` fails. But it failing with server crash with following error. That needs to be handled properly. I will handle in following PR.
```
 FATAL  Error: Authentication method 'token_exchange' is already registered

 server crashed  with status code 1
```